recipes/build.rb

```diff
@@ -37,16 +37,37 @@
   action :sync
 end
 
+ruby_block 'set env variabled' do
+  block do
+    require 'json'
+
+    file = File.read(File.join(dd_agent_omnibus_dir, "release.json"))
+    releases = JSON.parse(file)
+
+    release_version = node['dd-agent-builder']['release_version']
+    if not releases.key?(release_version)
+      puts "Error: could not find release #{}: exiting"
+      return
+    end
+
+    vars = ['AGENT_BRANCH', 'OMNIBUS_RUBY_BRANCH', 'OMNIBUS_SOFTWARE_BRANCH',
+            'INTEGRATIONS_CORE_BRANCH', 'TRACE_AGENT_BRANCH']
+    vars.each do |v|
+      if not releases[release_version].key?(v)
+        puts "Error: missing '#{v}' variable from release.json (version #{release_version})"
+        return
+      end
+      puts "exporting #{v}=#{releases[release_version][v]}"
+      ENV[v] = releases[release_version][v]
+    end
+  end
+end
+
 omnibus_build 'datadog-agent' do
   project_dir dd_agent_omnibus_dir
   log_level :info
   live_stream true
   install_dir node['dd-agent-builder']['install_dir']
-  environment 'AGENT_BRANCH' => node['dd-agent-builder']['dd-agent_branch'],
-              'OMNIBUS_RUBY_BRANCH' => node['dd-agent-builder']['omnibus-ruby_branch'],
-              'OMNIBUS_SOFTWARE_BRANCH' => node['dd-agent-builder']['omnibus-software_branch'],
-              'INTEGRATIONS_CORE_BRANCH' => node['dd-agent-builder']['integrations-core_branch'],
-              'TRACE_AGENT_BRANCH' => node['dd-agent-builder']['trace-agent_branch'],
-              'JMX_VERSION' => node['dd-agent-builder']['jmx-fetch_version']
+  environment 'JMX_VERSION' => node['dd-agent-builder']['jmx-fetch_version']
```

Comment:
do you know what this `JMX_VERSION` env var is for, and why we have it here but not in the `release.json` file?

Reply:
`JMX_VERSION` is compute in dd-agent-omnibus using git or curl if not set in the environment: https://github.com/DataDog/dd-agent-omnibus/blob/master/omnibus_build.sh#L56

Doing the same it out of the scope of this PR, but should be done at some point !

Reply:
ok thanks, hmm apparently the `JMX_VERSION` env var is never actually used on windows, see: https://github.com/DataDog/omnibus-software/blob/5.22.x/config/software/jmxfetch.rb#L4 : the version of jmxfetch is hardcoded in the software def.

Let's take this opportunity to clean up the mentions of `JMX_VERSION` and `jmx-fetch_version` from this recipe and the default attributes file? :)